loadtesting/proxy/tests/scheduled_tx.py

```diff
@@ -1,38 +1,158 @@
+        evm_loader.send_sol(bank_account, solana_account.pubkey(), int(1 * LAMPORT_PER_SOL))
+    else:
+        evm_loader.request_airdrop(solana_account.pubkey(), 1 * LAMPORT_PER_SOL)
+
+    # create owner
+    eth_account = account_manager.create_account()
+
+    # deploy a new erc20 contract
+    LOG.info("Start to deploy a contract...")
+    symbol = "".join([random.choice(string.ascii_uppercase) for _ in range(3)])
+    erc20 = ERC20NewWrapper(
+        web3_client,
+        faucet,
+        f"Test {symbol}",
+        symbol,
+        evm_loader,
+        solana_account=solana_account,
+        mintable=True,
+        bank_account=None,
+        account=eth_account,
+    )
+
+    LOG.info("Mint tokens...")
+    erc20.mint_tokens(signer=erc20.account, to_address=erc20.account.address, amount=10**18)
+
+    environment.contract_info["erc20_address"] = erc20.contract.address
+    environment.contract_info["erc20_owner_address"] = erc20.account.address
+
+    LOG.info("Create neon users...")
+    environment.contract_info["accounts"] = []
+    environment.contract_info["recipients"] = []
+    for i in range(neon_users):
+        neon_solana_account = Keypair()
+        LOG.info(f"Creating {i} neon user...")
+        neon_user = NeonUser(evm_loader.loader_id, keypair=neon_solana_account)
+        environment.contract_info["accounts"].append(neon_user)
+        if not network_object["use_bank"]:
+            evm_loader.request_airdrop(
+                neon_user.solana_account.pubkey(), 5 * LAMPORT_PER_SOL, commitment=commitment.Confirmed
+            )
+        else:
+            evm_loader.send_sol(bank_account, solana_account.pubkey(), int(5 * LAMPORT_PER_SOL))
+        LOG.info(f"Pop up {i} neon user balance...")
+        erc20.pop_up_balance(
+            evm_loader, recipient=neon_user, pda_amount=neon_user_balance, ata_amount=neon_user_balance
+        )
+        erc20.approve(erc20.account, neon_user.checksum_address, neon_user_balance)
 
+    environment.contract_info["recipients"] = environment.contract_info["accounts"].copy()
 
-@tag("scheduled_tx independent")
-class ScheduledTxsIndependentTasksSet(NeonProxyTasksSet):
-    """Implements independent scheduled txs sending pipeline tasks"""
+
+class BaseScheduledTxTaskSet(NeonProxyTasksSet):
 
     def on_start(self) -> None:
         super().on_start()
         super().setup()
         self.log = logging.getLogger("neon-consumer[%s]" % self.account.address[-8:])
 
+        with USER_LOCK:
+            if not self.user.environment.contract_info["accounts"]:
+                raise RuntimeError("Too little users")
+            self.neon_account = self.user.environment.contract_info["accounts"].pop(0)
+
+    def on_stop(self):
+        if self.account is not None:
+            with USER_LOCK:
+                self.user.environment.contract_info["accounts"].append(self.neon_account)
+                LOG.info(f"Returned user: {self.account.address}")
```

Comment:
in case of devnet run we have to add bank_account refund after every run. 

we have to collect sols from every neon_user.solana_account and return it to bank_account, you should use solana_client method drain_sol here https://github.com/neonlabsorg/neon-tests/blob/develop/utils/solana_client.py#L225 and also you can see an implemented refund mechanism in fixture https://github.com/neonlabsorg/neon-tests/blob/develop/integration/tests/conftest.py#L123 after yield

Reply:
sry, my comment is not in place here, we should add @events.test_stop.add_listener not on_stop hook, refund action should be performed once after the whole load test is finished

Reply:
why you don't use prepared method drain_sol of  environment.evm_loader
And if logic should be :

if network_object["use_bank"]: